dlt/destinations/impl/filesystem/filesystem.py

```diff
@@ -520,6 +520,31 @@ def _get_state_file_name(self, pipeline_name: str, version_hash: str, load_id: s
             f"{pipeline_name}{FILENAME_SEPARATOR}{load_id}{FILENAME_SEPARATOR}{self._to_path_safe_string(version_hash)}.jsonl",
         )
 
+    def _cleanup_pipeline_states(self, pipeline_name: str) -> None:
```

Comment:
In the ticket it says that we should make sure to not delete state files attached to failed loads, but we are not saving state on failed loads, so we should be good here.

Reply:
Yes, indeed. @rudolfix specified:

```
delete only the state files that corresponds to finished loads (they have corresponding completed entry). 
this is to prevent a rate case when we have 100 unsuccessful partial loads and we delete the last right state
```

will partial loads store a state file?